### ISSUE TITLE
[BUG] missing vanity short links

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/compiler-explorer#2582]
Apr 06 21:11:43 ip-172-30-0-113 amazon warn: Could not expand acm19_count4: Error: Missing attributes on acm19_count4 

(this is one of my ACM article links...I probably screwed it up but need to check)

**Comment by mattgodbolt** [comment]
Looks like all these "vanity" short links are gone

**Comment by mattgodbolt** [comment]
They're all there, but when we changed `MIN_STORED_ID_LENGTH` we broke them :)

```js
    getKeyStruct(id) {
        return {
            prefix: {S: id.substring(0, MIN_STORED_ID_LENGTH)},
            unique_subhash: {S: id},
        };
    }
```

... when searching from (e.g.) `acm19_count4` we used to search for a prefix of `acm19_` (six chars). We now look for 9 chars, which doesn't match anything.

**Comment by apmorton** [comment]
This is fixed by a194b5a591658d69eba5a9f12d7b5e596d82c561
